DiegoFlores/SpringCRUD/src/main/java/com/bootcamp/vetPatitas/service/PetServiceImp.java

```diff
@@ -0,0 +1,74 @@
+package com.bootcamp.vetPatitas.service;
+
+import static com.bootcamp.vetPatitas.utils.Constants.PET_INVALID_PHONE;
+import static com.bootcamp.vetPatitas.utils.Constants.PET_NULL_NAME;
+import static com.bootcamp.vetPatitas.utils.Constants.PET_NULL_TYPE;
+import static com.bootcamp.vetPatitas.utils.Constants.PET_NULL_AGE;
+import static com.bootcamp.vetPatitas.utils.Constants.PET_NULL_OWNER;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.bootcamp.vetPatitas.dao.PetDAO;
+import com.bootcamp.vetPatitas.dto.PetDTO;
+import com.bootcamp.vetPatitas.exception.InvalidDataException;
+import com.bootcamp.vetPatitas.mapper.PetMapper;
+
+
+@Service
+public class PetServiceImp implements PetService{
+
+	@Autowired
+	PetMapper petMapper;
+	
+	@Autowired
+	PetDAO petDao;
+	
+	@Override
+	public PetDTO addPet(PetDTO petDTO) {
```

Comment:
great!! you validate all input data

PS. Maybe you would like to read:
https://www.baeldung.com/javax-validation